packages/flutter/test/cupertino/nav_bar_transition_test.dart

```diff
@@ -59,26 +59,24 @@ Future<void> startTransitionBetween(
 }
 
 CupertinoPageScaffold? scaffoldForNavBar(Widget? navBar) {
-  if (navBar is CupertinoNavigationBar || navBar == null) {
-    return CupertinoPageScaffold(
-      navigationBar: navBar as CupertinoNavigationBar? ?? const CupertinoNavigationBar(),
-      child: const Placeholder(),
-    );
-  } else if (navBar is CupertinoSliverNavigationBar) {
-    return CupertinoPageScaffold(
-      child: CustomScrollView(
-        slivers: <Widget>[
+  switch (navBar) {
```

Comment:
Could this be a switch expression?

Reply:
You're a man of great taste: I'm currently on my [11th PR](https://github.com/flutter/flutter/pull/144580) in an attempt to refactor the entire repo to use switch expressions.

I like having each switch expression case as a single line, but I guess that's just a personal preference.

The real issue here is the default case: throwing an error works fine, but `assert(false); return null;` sadly can't be done.

Reply:
Ah right, missed that. Resolving!